idea/idea-gradle/src/org/jetbrains/kotlin/idea/configuration/GradleKotlinFrameworkSupportProvider.kt

```diff
@@ -187,21 +187,59 @@ open class GradleKotlinJavaFrameworkSupportProvider(
     override fun getDescription() = "A Kotlin library or application targeting the JVM"
 }
 
-open class GradleKotlinJSFrameworkSupportProvider(
-    frameworkTypeId: String = "KOTLIN_JS",
-    displayName: String = "Kotlin/JS"
+abstract class GradleKotlinJSFrameworkSupportProvider(
+    frameworkTypeId: String,
+    displayName: String
 ) : GradleKotlinFrameworkSupportProvider(frameworkTypeId, displayName, KotlinIcons.JS) {
+    abstract val jsSubTargetName: String
+
+    override fun addSupport(
+        buildScriptData: BuildScriptDataBuilder,
+        module: Module,
+        sdk: Sdk?,
+        specifyPluginVersionIfNeeded: Boolean,
+        explicitPluginVersion: String?
+    ) {
+        super.addSupport(buildScriptData, module, sdk, specifyPluginVersionIfNeeded, explicitPluginVersion)
+
+        buildScriptData.addOther(
+            """
+            kotlin {
+                target {
+                    $jsSubTargetName {
+                    }
+                }
+            }""".trimIndent()
```

Comment:
Minor: is there a reason to emit six lines with nested scopes instead of one or two (`kotlin.target.browser`)?

Reply:
User also may want to add `npm` dependencies into `sourceSets.main.depdencies`. But currently we generate `dependency` section above anyway. So I agree that shortcut is more preferable for now. Will fix.

Reply:
Fixed